src/libkernel/sscp/amdgpu/atomic_gfx90a.cpp

```diff
@@ -0,0 +1,24 @@
+/*
+ * This file is part of AdaptiveCpp, an implementation of SYCL and C++ standard
+ * parallelism for CPUs and GPUs.
+ *
+ * Copyright The AdaptiveCpp Contributors
+ *
+ * AdaptiveCpp is released under the BSD 2-Clause "Simplified" License.
+ * See file LICENSE in the project root for full license details.
+ */
+// SPDX-License-Identifier: BSD-2-Clause
+#include "hipSYCL/sycl/libkernel/sscp/builtins/atomic.hpp"
+#include "hipSYCL/sycl/libkernel/sscp/builtins/builtin_config.hpp"
+
+HIPSYCL_SSCP_BUILTIN __acpp_f32 __acpp_sscp_unsafe_atomic_fetch_add_f32(
+  __acpp_sscp_address_space as, __acpp_sscp_memory_order order,
+  __acpp_sscp_memory_scope scope, __acpp_f32 *ptr, __acpp_f32 x) {
+  return __builtin_amdgcn_global_atomic_fadd_f32( (const __attribute__((address_space(1))) float *)ptr, x);
+}
+
+HIPSYCL_SSCP_BUILTIN __acpp_f64 __acpp_sscp_unsafe_atomic_fetch_add_f64(
+  __acpp_sscp_address_space as, __acpp_sscp_memory_order order,
+  __acpp_sscp_memory_scope scope, __acpp_f64 *ptr, __acpp_f64 x) {
+  return __builtin_amdgcn_global_atomic_fadd_f64( (const __attribute__((address_space(1))) float *)ptr, x);
```

Comment:
I don't like this extra file - this will not scale (see ROCm device libs...). Why can we not solve this with reflection in the existing file? I assume because clang only generates the right IR for the builtin when it sees the right target?
Inline assembly + reflection might still be easier...